code/modules/organs/organ_external.dm

```diff
@@ -610,18 +610,9 @@ Note that amputating the affected organ does in fact remove the infection from t
 	var/datum/species/species = src.species || owner.species
 
 	for(var/datum/wound/W in wounds)
-		//Internal wounds get worse over time. Low temperatures (cryo) stop them.
-		if(W.internal && !W.is_treated() && owner.bodytemperature >= 170 && !(species && species.anatomy_flags & NO_BLOOD))
-			var/blood_factor = 1
-
-			if(owner.reagents.has_reagent(INAPROVALINE) || owner.reagents.has_reagent(BICARIDINE)) //Inaprovaline and bicaridine slow bleeding by 30%
-				blood_factor -= 0.3
-
-			if(owner.reagents.has_reagent(HYPERZINE)) //On the other hand, hyperzine speeds up bleeding by 30%
-				blood_factor += 0.3
-
-			if(owner.reagents.has_reagent(CLOTTING_AGENT) || owner.reagents.has_reagent(BIOFOAM)) //Clotting agent and biofoam stop bleeding entirely.
-				blood_factor = 0
+		//Internal wounds get worse over time, and you lose blood
+		if(W.internal && !W.is_treated() && !(species && species.anatomy_flags & NO_BLOOD))
+			var/blood_factor = owner.calcbloodloss() //Calls helper function to determine amount of blood loss.
 
 			owner.vessel.remove_reagent(BLOOD, W.damage * wound_update_accuracy * 0.07 * max(0, blood_factor))
```

Comment:
I was gonna suggest that this be done in an unrelated PR, but since these reagents checks are getting removed, it seems appropriate for this scope to re-implement wounds slowly opening more and more.
see [these three removed lines](https://github.com/vgstation-coders/vgstation13/pull/28915/files#diff-a1ca7744b58972eda50448593079d96db87072abec59d0d5780bcc99219d9c35L617) for how it used to work before hyperzine-bleeding was added.

Reply:
and idk how necessary or in-scope this is, but it might also be nice to just have blood_factor be a variable on wounds, and then have these three lines instead read `W.blood_factor`
```
for(var/datum/wound/W in temp.wounds)
	if(W.bleeding())
		blood_max += W.damage / 4
```
in however a manner you choose, I suppose - whether they add to `blood_factor` or `blood_max` in handle_blood()

---
or in other words, something like this:

`code/modules/organs/wound.dm`
```
/datum/wound
  var/blood_factor = 1
```
`code/modules/organs/organ_external.dm`
```
/datum/organ/external/proc/update_wounds()
	...
	for(var/datum/wound/W in wounds)
		if(W.internal && !W.is_treated() && !(species && species.anatomy_flags & NO_BLOOD))
			blood_factor = owner.calcbloodloss() //Calls helper function to determine amount of blood loss.
		...
		// Wound-opening thing from comment above.
		if(!owner.reagents.has_any_reagents(list(BICARIDINE,INAPROVALINE,CLOTTING_AGENT,BIOFOAM)))	//Bicard, inaprovaline, clotting agent, and biofoam stop internal wounds from growing bigger with time, and also slow bleeding
			W.open_wound(0.1 * wound_update_accuracy)
			owner.vessel.remove_reagent(BLOOD, 0.05 * W.damage * wound_update_accuracy)
```
`code/modules/organs/blood.dm`
```
/mob/living/carbon/human/proc/handle_blood()
	...
	for(var/datum/wound/W in temp.wounds)
		if(W.bleeding())
			blood_max += W.blood_factor
```
or whatever, idk. would just be nice if there weren't 2 sources of on-Life bloodloss, or if the other one was only for wounds opening.

Reply:
Response to your first comment is that it would be a different PR, it's a bit unatomic to unnerf something like the wounds-not-actually-expanding here.

Second comment's response is that wound code is really jank code. I don't want to go in depth here but:

> **blood.dm**
> handles all externally bleeding wounds
> bleeding() returns 0 if the wound is internal, so they aren't handled here at all

> **organ_external.dm**
> handles all "internal wounds" (wounds inside of external organs vs outside of external organs)
> internal wounds are all called "internal bleeding" but do not bleed normally as above
> internal wounds bleed as they expand
> due to an older PR, wounds no longer actually expand, but the blood loss is still there

>would just be nice if there weren't 2 sources of on-Life bloodloss, or if the other one was only for wounds opening.

Wounds opening are the only way internal wounds bleed, so, basically, this is already done.

Also, blood_max is maximum amount of blood loss (in units), blood_factor is the scaling of this blood loss based on the factors listed in the PR (so your blood loss formula would be something like bleed(blood_max * blood_factor). This PR basically organizes the blood_factor calculations into a single function that's only called when bleeding occurs. Saving the result in the wound itself is a waste of memory since the result can change every life() tick but only needs to be calculated when bleeding is happening.

I think we're good here, minus a refactoring of wound code, and the above re-adding of the expanding wounds code.

Reply:
yeah but it _isn't_ in a single function. bloodloss happens in `handle_blood()` due to the eventual drip() call, and it also currently happens in update_wounds() due to the remove_reagent() call. they're distinct and separate for no good reason I can see, and the latter straight up just removes a volume of `BLOOD` from the vessel.
the reason the latter existed in the first place _was_ because of open_wounds(), but open_wounds() was removed and remove_reagent() is still there.

Reply:
Handled on Discord(tm)

Tl;dr update_wounds() ONLY handles internal wounds.